scipy/stats/morestats.py

```diff
@@ -1121,9 +1154,48 @@ def boxcox_normmax(x, brack=(-2.0, 2.0), method='pearsonr'):
 
     >>> plt.show()
 
+    Alternatively, we can define our own `optimizer` function. Suppose we
+    are only interested in values of `lmbda` on the interval [6, 7], we
+    want to use `scipy.optimize.minimize_scalar` with ``method='bounded'``,
+    and we want to use tighter tolerances when optimizing the log-likelihood
+    function. To do this, we define a function that accepts positional argument
+    `fun` and uses `scipy.optimize.minimize_scalar` to minimize `fun` subject
+    to the provided bounds and tolerances:
+
+    >>> from scipy import optimize
+    >>> options = {'xatol': 1e-12}  # absolute tolerance on `x`
+    >>> def optimizer(fun):
+    ...     return optimize.minimize_scalar(fun, bounds=(6, 7),
+    ...                                     method="bounded", options=options)
+    >>> stats.boxcox_normmax(x, optimizer=optimizer)
+    6.999...
     """
+    # If optimizer is not given, define default 'brent' optimizer.
+    if optimizer is None:
+
+        # Set default value for `brack`.
+        if brack is None:
+            brack = (-2.0, 2.0)
+
+        def _optimizer(func, args):
+            return optimize.brent(func, args=args, brack=brack)
+
+    # Otherwise check optimizer.
+    else:
+        if not callable(optimizer):
+            raise ValueError("`optimizer` must be a callable")
+
+        if brack is not None:
+            raise ValueError("`brack` must be None if `optimizer` is given")
```

Comment:
This is not tested and incorrect with the description of the parameter. Either remove this check or update the doc and add a test for this case.

Reply:
Instead of `must be None` it would be accurate to say `will be ignored`, right?

Reply:
Yes but then it should not raise but just emit a warning. Is this what you suggest? Works for me too, it would be coherant.

Reply:
I'm not sure what the general policy is in SciPy, but if I specify a parameter (i.e. `brack`) explicitly but it is not used, I would prefer an error rather than a warning. I'm happy to update the docs and add a test to check that the error is raised if `brack` and `optimizer` is given.

Reply:
I too prefer the error than the warning. A lot of people don't read those... (cf. tensorflow and such which are killing logs...)

Reply:
I actually meant no warning, no error, just ignored, but this is fine, too, and better for users.